src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -108,7 +108,8 @@ private function findClasses($namespace, $pattern)
             if (!$r = $this->container->getReflectionClass($class)) {
                 throw new InvalidArgumentException(sprintf('Expected to find class "%s" in file "%s" while importing services from resource "%s", but it was not found! Check the namespace prefix used with the resource.', $class, $path, $pattern));
             }
-            if (!$r->isInterface() && !$r->isTrait()) {
+            // skip interfaces, traits, abstract classes & private constructors
+            if ($r->isInstantiable()) {
```

Comment:
I'm not certain we should skip private constructors, users could define a factory when having some.